app/src/auth/components/EthereumLogin/index.jsx

```diff
@@ -1,30 +1,48 @@
 // @flow
 
-import React from 'react'
+import React, { useContext, useRef, useEffect, useCallback } from 'react'
 import { I18n } from 'react-redux-i18n'
 
+import { type Ref } from '$shared/flowtype/common-types'
+import AuthFormProvider from '../AuthFormProvider'
+import AuthFormContext from '../../contexts/AuthForm'
+import SessionContext from '../../contexts/Session'
 import AuthPanel from '$auth/components/AuthPanel'
 import AuthStep from '$auth/components/AuthStep'
 import TextInput from '$shared/components/TextInput'
 import getSessionToken from '$auth/utils/getSessionToken'
 import { getWeb3 } from '$shared/web3/web3Provider'
-import { type Props as SessionProps } from '$auth/contexts/Session'
-import { type AuthFlowProps } from '$shared/flowtype/auth-types'
 
-type Props = SessionProps & AuthFlowProps & {
-    form: {
-        ethereum: null,
-    },
+type Props = {
     onBackClick: () => void,
 }
 
-class EthereumLogin extends React.Component<Props> {
-    componentWillUnmount() {
-        this.unmounted = true
-    }
+type Form = {
+    ethereum: any,
+}
+
+const initialForm: Form = {
+    ethereum: null,
+}
+
+const EthereumLogin = ({ onBackClick }: Props) => {
+    const {
+        errors,
+        isProcessing,
+        redirect,
+        setFieldError,
+        step,
+    } = useContext(AuthFormContext)
+
+    const mountedRef: Ref<boolean> = useRef(true)
 
-    submit = async () => {
-        const { setSessionToken, setFieldError } = this.props
+    useEffect(() => () => {
+        mountedRef.current = false
+    }, [])
```

Comment:
Could probably factor this hook out since it's highly reusable. e.g. 

```js
function useIsMounted() {
    const mountedRef = useRef(true)
    useEffect(() => () => {
        mountedRef.current = false
    }, [])
   return mountedRef
}

// …
const isMounted = useIsMounted()
if (isMounted.current) {

}
```

Reply:
Named it `useIsMountedRef` cause technically it's still a ref.

Reply:
agreed, this is better.